### PR TITLE
Provide PHP 8.1 support

### PR DESCRIPTION
**Pull request by weierophinney**
This patch provides PHP 8.1 support, via the following changes:

- Updates the PHP constraint to add `~8.1.0`
- Updates the way the package replaces zend-mime
  - Replaces the "replace" section of the package with "conflict", and conflicts with all versions of zend-mime
  - Removes the dependency on laminas-zendframework-bridge
- Updates to and applies laminas-coding-standard 2.2 rules
- Updates testing to inline the `Headers` class

> We have a circular dependency scenario between laminas-mail and laminas-mime.  laminas-mail depends on laminas-mime for its MIME parsing capabilities.  laminas-mime decodes headers to a laminas-mail `Headers` instance.
>
> Unfortunately the `Headers` class implements `Iterator`, and, as such, we get deprecations under 8.1 unless the `#[ReturnTypeWillChange]` attribute to affected methods.
>
> The solution here is to temporarily inline the `Headers` class as a test asset until laminas-mail is updated to work on PHP 8.1.

### CODE REVIEW

File: composer.lock
Comment (Ocramius) [inline composer.lock:526]: `laminas/laminas-zendframework-bridge` still made it in here: let's try getting rid of it before merge

**Review by Ocramius — APPROVED** [review]

File: composer.lock
Comment (weierophinney) [reply composer.lock:526]: We can't - it's from laminas-mail. So until both have their release for 8.1, this one will have the bridge. It'll have to wait until the next minor after this one, or next patch release, if we update the lockfile then.

Comment (Ocramius) [reply composer.lock:526]: got it, makes sense

**Pull request merged**
